dpctl/_sycl_queue.pyx

```diff
@@ -121,6 +122,86 @@ cdef class kernel_arg_type_attribute:
         return self.attr_value
 
 
+cdef class LocalAccessor:
+    """
+    LocalAccessor(ndim, dtype, dim0, dim1, dim2)
+
+    Python class for specifying the dimensionality and type of a
+    ``sycl::local_accessor``, to be used as a kernel argument type.
+
+    Args:
+        ndim (size_t):
+            number of dimensions.
+            Can be between one and three.
+        dtype (str):
+            the data type of the local memory.
+            The permitted values are
+
+                `'i1'`, `'i2'`, `'i4'`, `'i8'`:
+                    signed integral types int8_t, int16_t, int32_t, int64_t
+                `'u1'`, `'u2'`, `'u4'`, `'u8'`
+                    unsigned integral types uint8_t, uint16_t, uint32_t,
+                    uint64_t
+                `'f4'`, `'f8'`,
+                    single- and double-precision floating-point types float and
+                    double
+        dim0 (size_t):
+            Size of the first dimension.
+        dim1 (size_t):
+            Size of the second dimension.
+        dim2 (size_t):
+            Size of the third dimension.
```

Comment:
Does a user always need to specify all three dimensions? Could we enable them to only specify as many as needed and have the others default to `1`?

The best way to achieve that would probably be to take a list of 1, 2 or 3 elements and use them similar to a `range` in SYCL.

Reply:
numba-dpex also chose to use a tuple/list [when it wrapped `local_accessor`](https://github.com/IntelPython/numba-dpex/blob/72f3850c88b4a3e2e21f589bda094e3ee4d609f7/numba_dpex/kernel_api/local_accessor.py#L31)

I think it would be sensible to do something similar

Reply:
This has been implemented.